### PR TITLE
Add policy to highlight Linked Edges and Adjacent Nodes by selecting node.

### LINKED ISSUE #9036: [FEATURE] New Graph Selection/Inspection Policy: Node, Linked Edges, and Linked Nodes
I am currently using the `NodesAndLinkedEdges()` policy for my graph inspection policy and selection policy. Currently, a node is selected, and it becomes highlighted along with every edge leading in or out of that node. **However, I want every node attached to the highlighted edges to also be highlighted.**  

If someone could help clarify where exactly the NodesAndLinkedEdges() class is actually implemented, I would consider doing this myself, as my feature request is an extension from that class. 

It could be called `NodesLinkedEdgesandLinkedNodes()`.

### PR DESCRIPTION
**Pull request by axie1234**
The NodesAndLinkedEdgesAndLinkedNodes policy highlights every edge leading in and out of a selected node, as well as highlights every adjacent node that is connected with the adjacent nodes. A single selection is highlighted with all connected edges adjacent nodes connected with the edges, with the selected node as the last entry to identify it from its adjacent nodes.  If a selected node does not have any edges, only that node is highlighted in the selection.

- [x] issues: fixes #9036 
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by g-parki** [comment]
@axie1234 the CI run is currently failing due to whitespace errors (see output below). Ideally these should be caught on your machine prior to pushing. Here are some options to help clean up:

- Run codebase tests manually: `pytest tests/codebase`
- [Set up pre-commit](https://docs.bokeh.org/en/latest/docs/dev_guide/setup.html#set-up-pre-commit) to run tests automatically before pushing
- See full details for Bokeh tests in the [dev guide](https://docs.bokeh.org/en/latest/docs/dev_guide/testing.html)


```
E       AssertionError: Code quality issues:
E         File contains trailing whitespace: bokeh/models/graphs.py, line 182.
E         File contains trailing whitespace: bokeh/models/graphs.py, line 183.
E         File contains trailing whitespace: bokeh/models/graphs.py, line 184.
E         File contains trailing whitespace: bokeh/models/graphs.py, line 185.
E         File contains trailing whitespace: examples/models/file/graphs.py, line 15.
E         File contains trailing whitespace: examples/models/file/graphs.py, line 16.
```

**Comment by axie1234** [comment]
@g-parki Will do. I have exams in the next few days. My partner for this project will probably take a look at it.

**Comment by Pranav143** [comment]
Visualization of this new inspection policy. Hovering over a node now highlights all connected nodes as well as the edges going out to those nodes

task 4 pic

File: bokeh/models/graphs.py [hunk 1]
```diff
@@ -174,6 +175,19 @@ class NodesAndAdjacentNodes(GraphHitTestPolicy):
...
 
+class NodesAndLinkedEdgesAndLinkedNodes(GraphHitTestPolicy):
+    '''
+    With the ``NodesAndLinkedEdgesAndLinkedNodes`` policy, inspection or selection
+    of graph nodes will result in the inspection or selection of the node, any their
+    linked graph edges, and of the linked graph nodes. There is no selection or
+    inspection of graph edges, and no indication of which node is the tool-selected
+    one from the policy-selected nodes. There is no selection or inspection of graph
+    edges, and no indication of which node is the tool-selected one from the
+    policy-selected nodes.
```
Comment (g-parki) [inline bokeh/models/graphs.py:186]:
```suggestion
    With the ``NodesAndLinkedEdgesAndLinkedNodes`` policy, inspection or selection
    of graph nodes will result in the inspection or selection of the node, any
    linked graph edges, and any adjacent graph nodes. There is no selection or
    inspection of graph edges, and no indication of which node is tool-selected.
```
Suggestion to bring down the word count and use "adjacent" to remain consistent with my other comment of renaming to `NodesAndLinkedEdgesAndAdjacentNodes`

**Comment by g-parki** [comment]
@axie1234 @Pranav143 thanks for the work so far on this PR! I left a couple suggestions to get the conversation started on which direction this should go. Also I noticed the `package-lock.json` was committed to this branch, which will need to be removed before this PR can be considered for merging.

**Comment by bryevdv** [comment]
FYI I'm going to open a discussion about a larger design revamp but I'm in the middle of a move, so it will be a few days.

**Comment by bryevdv** [comment]
Sorry I am obviously way behind on this, but  I have not forgotten it.

**Comment by bryevdv** [comment]
@axie1234 @g-parki @bokeh/dev the discussion is started here 

* https://github.com/bokeh/bokeh/discussions/11952

**Comment by bryevdv** [comment]
@axie1234 thank you very much for the PR submission.  Now that there are conflicts, I am going to go ahead and close this PR.  I am not sure when the discussion and work aroundit will pickup, but in the mean time, I just can't justify adding new code under the current unsustainable scheme. However, I do think the correct discussion has been started as a result, which will help us get to the right place eventually.

**Pull request closed without merging**